src/Symfony/Component/DependencyInjection/Loader/IniFileLoader.php

```diff
@@ -52,7 +52,15 @@ public function load($resource, $type = null)
      */
     public function supports($resource, $type = null)
     {
-        return is_string($resource) && 'ini' === pathinfo($resource, PATHINFO_EXTENSION);
+        if (!is_string($resource)) {
+            return false;
+        }
+
+        if (null === $type && 'ini' === pathinfo($resource, PATHINFO_EXTENSION)) {
+            return true;
+        }
+
+        return null !== $type && 'ini' === $type;
```

Comment:
> the loader won't return true in case the type was explicitly provided, but not ini and the extension is ini (another loader should try)

@ogizanagi in that case, this code will be:

```php
return 'ini' === $type || 'ini' === pathinfo($resource, PATHINFO_EXTENSION);
return false || true;
return true;
```

Maybe I'm missing something here.

@stof maybe "priority" is not the best word, but the new behavior is: *"do whatever "type" tells you ... if it doesn't tell you anything, fall back to the file extension"*.